isis/tests/FunctionalTestsJigsaw.cpp

```diff
@@ -143,4 +145,97 @@ TEST_F(ObservationPair, FunctionalTestJigsawCamSolveAll) {
   // DEC(t3) final
   EXPECT_NEAR(elems.at(56).toDouble(), 0.365717165,          0.00001); 
   
-}
\ No newline at end of file
+}
+
+TEST_F(ObservationPair, FunctionalTestJigsawErrorNoSolve) {
+  // delete to remove old camera for when cam is updated
+  QTemporaryDir prefix;
+  QString outCnetFileName = prefix.path() + "/outTemp.net";
+  QVector<QString> args = {"fromlist="+cubeListFile, "cnet="+cnetPath, "onet="+outCnetFileName, 
+                           "camsolve=None", "spsolve=None"};
+
+  UserInterface options(APP_XML, args);
+  
+  Pvl log; 
+  
+  try {
+    std::cout << "running" << std::endl;
+    jigsaw(options, &log);
+    FAIL() << "Should throw" << std::endl;
+  }
+  catch (IException &e) {
+    EXPECT_THAT(e.what(), HasSubstr("Must either solve for camera pointing or spacecraft position"));
+  }
+}
+
+
+TEST_F(ObservationPair, FunctionalTestJigsawErrorNoNet) {
+  QVector<QString> args = {"fromlist="+cubeListFile, "cnet=lolfake.net", "onet=doesnotmatter"};
+
+  UserInterface options(APP_XML, args);
+  
+  Pvl log; 
+  
+  try {
+    jigsaw(options, &log);
+    FAIL() << "Should throw an exception" << std::endl;
+  }
+  catch (IException &e) {
+    EXPECT_THAT(e.what(), HasSubstr("Unable to open"));
+  }
+}
+
+
+TEST_F(ObservationPair, FunctionalTestJigsawErrorTBParamsNoTarget) {
+  QTemporaryDir prefix;
+  QString outCnetFileName = prefix.path() + "/outTemp.net";
+  
+  // just use isdPath for a valid PVL file without the wanted groups
+  QVector<QString> args = {"fromlist="+cubeListFile, "cnet="+cnetPath, "onet="+outCnetFileName, "SOLVETARGETBODY=TRUE", "tbparameters="+cubeRPath};
+
+  UserInterface options(APP_XML, args);
+  
+  Pvl log; 
+  
+  try {
+    jigsaw(options, &log);
+    FAIL() << "Should throw an exception" << std::endl;
+  }
+  catch (IException &e) {
+    EXPECT_THAT(e.what(), HasSubstr("Input Target parameters file missing main Target object"));
+  } 
+}
+
+
+TEST_F(ObservationPair, FunctionalTestJigsawErrorTBParamsNoSolve) {
+  QTemporaryDir prefix;
+  QString outCnetFileName = prefix.path() + "/outTemp.net";
+  
+  std::istringstream iss(R"(
+    Object = Target
+    Group = "NAME"
+       Name=Enceladus
+    EndGroup
+    END_OBJECT
+  )"); 
+  
+  QString tbsolvepath = "/tmp/tbsolve.pvl";
```

Comment:
Any particular reason to write directly to this file in /tmp/ rather than putting it in `tempDir.path()`? My preference would be to put it in that directory, just to keep things organized between test runs.

Reply:
also a debug things, need to revert to prefix, basically for checking the file to make sure everything looks good before doing asserts.